sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/validation/CapabilityValidator.java

```diff
@@ -20,6 +20,15 @@ public CapabilityValidator(Capabilities capabilities) {
     }
 
     public void validate(ServiceSpec serviceSpec) throws CapabilityValidationException {
+        // GPU_RESOURCE is a framework level option, so we check for compatibility here
```

Comment:
Can you please break this code into it's own protected method ? And add a test ?